Check datasets against CDC website to ensure imported correctly

In [29]:
# Dependencies
from pathlib import Path
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as st
from scipy.stats import linregress
import pyreadstat
import xport

In [30]:
# Filtered_Renamed_Data import 
Check_dataset = pd.read_csv("Datasets/Filtered_Renamed_Data")
Check_dataset

Respondent ID  Taken Rx Past Month  Number Days Taken Rx  \
0           109263.0                  2.0                   NaN   
1           109264.0                  2.0                   NaN   
2           109265.0                  2.0                   NaN   
3           109266.0                  2.0                   NaN   
4           109267.0                  1.0                   NaN   
...              ...                  ...                   ...   
32957       124821.0                  1.0                2555.0   
32958       124821.0                  1.0                 730.0   
32959       124821.0                  1.0                3650.0   
32960       124822.0                  1.0                1460.0   
32961       124822.0                  1.0                1460.0   

       Number of Rx Meds  Gender  Age at Screening  Race  \
0                    NaN     1.0               2.0   6.0   
1                    NaN     2.0              13.0   1.0   
2                    NaN     1.0               2.0   3.0   
3                    NaN     2.0              29.0   6.0   
4                    1.0     2.0              21.0   2.0   
...                  ...     ...               ...   ...   
32957                4.0     1.0              63.0   4.0   
32958                4.0     1.0              63.0   4.0   
32959                4.0     1.0              63.0   4.0   
32960                2.0     1.0              74.0   2.0   
32961                2.0     1.0              74.0   2.0   

       Educational Attainment  Marital Status  Sample Interview Weight  ...  \
0                         NaN             NaN              7891.762435  ...   
1                         NaN             NaN             11689.747264  ...   
2                         NaN             NaN             16273.825939  ...   
3                         5.0             3.0              7825.646112  ...   
4                         4.0             3.0             26379.991724  ...   
...                       ...             ...                      ...  ...   
32957                     2.0             2.0              4666.817952  ...   
32958                     2.0             2.0              4666.817952  ...   
32959                     2.0             2.0              4666.817952  ...   
32960                     3.0             1.0             14024.476634  ...   
32961                     3.0             1.0             14024.476634  ...   

       Bedtime Week/Workdays  Waketime Week/Workdays  \
0                        NaN                     NaN   
1                        NaN                     NaN   
2                        NaN                     NaN   
3                   b'22:00'                b'05:30'   
4                   b'00:00'                b'08:00'   
...                      ...                     ...   
32957               b'21:00'                b'04:45'   
32958               b'21:00'                b'04:45'   
32959               b'21:00'                b'04:45'   
32960               b'22:30'                b'04:00'   
32961               b'22:30'                b'04:00'   

       Sleep Hours Week/Workdays  Bedtime Weekends  Waketime Weekends  \
0                            NaN               NaN                NaN   
1                            NaN               NaN                NaN   
2                            NaN               NaN                NaN   
3                            7.5          b'23:00'           b'07:00'   
4                            8.0          b'03:00'           b'11:00'   
...                          ...               ...                ...   
32957                        8.0          b'00:00'           b'09:00'   
32958                        8.0          b'00:00'           b'09:00'   
32959                        8.0          b'00:00'           b'09:00'   
32960                        5.5          b'22:30'           b'06:30'   
32961                        5.5          b'22:30'      

In [31]:
Check_dataset.columns

Index(['Respondent ID', 'Taken Rx Past Month', 'Number Days Taken Rx',
       'Number of Rx Meds', 'Gender', 'Age at Screening', 'Race',
       'Educational Attainment', 'Marital Status', 'Sample Interview Weight',
       'MEC Sample Weight', 'Family Income to Poverty Ratio',
       'Vigorous Work Activity', 'Number of Days Vigorous Work',
       'Minutes of Vigorous Work Daily', 'Moderate Work Activity',
       'Number of Days Moderate Work', 'Minutes of Moderate Work Daily',
       'Walk or Bike Travel', 'Number of Days Walk or Bike Weekly',
       'Daily Minutes Walking or Biking', 'Vigorous Recreational Activity',
       'Number of Days Vigorous Recreation',
       'Minutes of Vigorous Recreation Daily',
       'Moderate Recreational Activity', 'Number of Days Moderate Recreation',
       'Minutes of Moderate Recreation Daily',
       'Minutes of Sedentary Activity Daily', 'Bedtime Week/Workdays',
       'Waketime Week/Workdays', 'Sleep Hours Week/Workdays',
       'Bedtime Weekend

In [32]:
#check Prescription Meds counts
rx_check = Check_dataset.drop(columns=['Gender', 'Age at Screening', 'Race',
       'Educational Attainment', 'Marital Status', 'Sample Interview Weight',
       'MEC Sample Weight', 'Family Income to Poverty Ratio',
       'Vigorous Work Activity', 'Number of Days Vigorous Work',
       'Minutes of Vigorous Work Daily', 'Moderate Work Activity',
       'Number of Days Moderate Work', 'Minutes of Moderate Work Daily',
       'Walk or Bike Travel', 'Number of Days Walk or Bike Weekly',
       'Daily Minutes Walking or Biking', 'Vigorous Recreational Activity',
       'Number of Days Vigorous Recreation',
       'Minutes of Vigorous Recreation Daily',
       'Moderate Recreational Activity', 'Number of Days Moderate Recreation',
       'Minutes of Moderate Recreation Daily',
       'Minutes of Sedentary Activity Daily', 'Bedtime Week/Workdays',
       'Waketime Week/Workdays', 'Sleep Hours Week/Workdays',
       'Bedtime Weekends', 'Waketime Weekends', 'Sleep Hours Weekends',
       'Snoring Frequency Past Year',
       'Frequency Snort/Stop Breathing Past Year',
       'Told Doctor About Sleep Issues', 'Feeling Tired During Day Frequency'
                                      ])
rx_check                                   
                     

Respondent ID  Taken Rx Past Month  Number Days Taken Rx  \
0           109263.0                  2.0                   NaN   
1           109264.0                  2.0                   NaN   
2           109265.0                  2.0                   NaN   
3           109266.0                  2.0                   NaN   
4           109267.0                  1.0                   NaN   
...              ...                  ...                   ...   
32957       124821.0                  1.0                2555.0   
32958       124821.0                  1.0                 730.0   
32959       124821.0                  1.0                3650.0   
32960       124822.0                  1.0                1460.0   
32961       124822.0                  1.0                1460.0   

       Number of Rx Meds  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4                    1.0  
...                  ...  
32957                4.0  
32958                4.0  
32959                4.0  
32960                2.0  
32961                2.0  

[32962 rows x 4 columns]

In [33]:
#check counts in each column
rx_check.count()

Respondent ID           32962
Taken Rx Past Month     32962
Number Days Taken Rx    24031
Number of Rx Meds       24246
dtype: int64

In [34]:
rx_check["Respondent ID"].unique()

array([109263., 109264., 109265., ..., 124820., 124821., 124822.])

In [35]:
#check unique respondents
rx_array = rx_check["Respondent ID"].unique()
len(rx_array)

15560

Unique SEQN entries = 15560 but total rows = 32962. 32962 is consistent with what is displayed for cumulative responses for rx information on CDC website. 

In reading further: "Respondent sequence number (SEQN) in the prescription medication data is not a unique identifier. Participants who reported the use of multiple prescription drugs will have multiple records, that is, a record for each drug. Analysts should convert a drug-level data to a person-level data, that is, a record for each person, before merging it with NHANES demographic and other data files by using SEQN."

So all merged data will have multiple entries due to the SEQN not being a unique identifier for rx data. 

Possible Solutions: 
1) eliminate duplicates, since the columns that we are interested in should have the same values (# of meds taken will be consistent, # days taken will not)
2) unmerge the data - take out meds and sleep, analyze separately given issues with duplicates and encoding

In [36]:
#based on reasoning above, removed dups from demographic data to check counts
demographics_check = Check_dataset.drop(columns=['Taken Rx Past Month', 'Number Days Taken Rx',
       'Number of Rx Meds', 'Vigorous Work Activity', 'Number of Days Vigorous Work',
       'Minutes of Vigorous Work Daily', 'Moderate Work Activity',
       'Number of Days Moderate Work', 'Minutes of Moderate Work Daily',
       'Walk or Bike Travel', 'Number of Days Walk or Bike Weekly',
       'Daily Minutes Walking or Biking', 'Vigorous Recreational Activity',
       'Number of Days Vigorous Recreation',
       'Minutes of Vigorous Recreation Daily',
       'Moderate Recreational Activity', 'Number of Days Moderate Recreation',
       'Minutes of Moderate Recreation Daily',
       'Minutes of Sedentary Activity Daily', 'Bedtime Week/Workdays',
       'Waketime Week/Workdays', 'Sleep Hours Week/Workdays',
       'Bedtime Weekends', 'Waketime Weekends', 'Sleep Hours Weekends',
       'Snoring Frequency Past Year',
       'Frequency Snort/Stop Breathing Past Year',
       'Told Doctor About Sleep Issues', 'Feeling Tired During Day Frequency'])
   
demographics_check

Respondent ID  Gender  Age at Screening  Race  Educational Attainment  \
0           109263.0     1.0               2.0   6.0                     NaN   
1           109264.0     2.0              13.0   1.0                     NaN   
2           109265.0     1.0               2.0   3.0                     NaN   
3           109266.0     2.0              29.0   6.0                     5.0   
4           109267.0     2.0              21.0   2.0                     4.0   
...              ...     ...               ...   ...                     ...   
32957       124821.0     1.0              63.0   4.0                     2.0   
32958       124821.0     1.0              63.0   4.0                     2.0   
32959       124821.0     1.0              63.0   4.0                     2.0   
32960       124822.0     1.0              74.0   2.0                     3.0   
32961       124822.0     1.0              74.0   2.0                     3.0   

       Marital Status  Sample Interview Weight  MEC Sample Weight  \
0                 NaN              7891.762435       8.951816e+03   
1                 NaN             11689.747264       1.227116e+04   
2                 NaN             16273.825939       1.665876e+04   
3                 3.0              7825.646112       8.154968e+03   
4                 3.0             26379.991724       5.397605e-79   
...               ...                      ...                ...   
32957             2.0              4666.817952       4.853430e+03   
32958             2.0              4666.817952       4.853430e+03   
32959             2.0              4666.817952       4.853430e+03   
32960             1.0             14024.476634       1.519580e+04   
32961             1.0             14024.476634       1.519580e+04   

       Family Income to Poverty Ratio  
0                                4.66  
1                                0.83  
2                                3.06  
3                                5.00  
4                                5.00  
...                               ...  
32957                            3.71  
32958                            3.71  
32959                            3.71  
32960                             NaN  
32961                             NaN  

[32962 rows x 9 columns]

In [37]:
demo_remove_dups = demographics_check.drop_duplicates(subset=['Respondent ID'], keep='last')
demo_remove_dups

Respondent ID  Gender  Age at Screening  Race  Educational Attainment  \
0           109263.0     1.0               2.0   6.0                     NaN   
1           109264.0     2.0              13.0   1.0                     NaN   
2           109265.0     1.0               2.0   3.0                     NaN   
3           109266.0     2.0              29.0   6.0                     5.0   
4           109267.0     2.0              21.0   2.0                     4.0   
...              ...     ...               ...   ...                     ...   
32953       124818.0     1.0              40.0   4.0                     5.0   
32954       124819.0     1.0               2.0   4.0                     NaN   
32955       124820.0     2.0               7.0   3.0                     NaN   
32959       124821.0     1.0              63.0   4.0                     2.0   
32961       124822.0     1.0              74.0   2.0                     3.0   

       Marital Status  Sample Interview Weight  MEC Sample Weight  \
0                 NaN              7891.762435       8.951816e+03   
1                 NaN             11689.747264       1.227116e+04   
2                 NaN             16273.825939       1.665876e+04   
3                 3.0              7825.646112       8.154968e+03   
4                 3.0             26379.991724       5.397605e-79   
...               ...                      ...                ...   
32953             1.0             21586.596728       2.166689e+04   
32954             NaN              1664.919253       1.838170e+03   
32955             NaN             14819.783161       1.649781e+04   
32959             2.0              4666.817952       4.853430e+03   
32961             1.0             14024.476634       1.519580e+04   

       Family Income to Poverty Ratio  
0                                4.66  
1                                0.83  
2                                3.06  
3                                5.00  
4                                5.00  
...                               ...  
32953                            3.82  
32954                            0.07  
32955                            1.22  
32959                            3.71  
32961                             NaN  

[15560 rows x 9 columns]

In [38]:
demo_remove_dups.count()

Respondent ID                     15560
Gender                            15560
Age at Screening                  15560
Race                              15560
Educational Attainment             9232
Marital Status                     9232
Sample Interview Weight           15560
MEC Sample Weight                 15560
Family Income to Poverty Ratio    13359
dtype: int64

Demographics data pass the check, as unique SEQN identifiers for this dataset = 15560 which is the reported cumulative number of responses for demographics data. This makes sense as gender was chosen for all respondents.

In [39]:
physical_activity_check = Check_dataset.drop(columns=['Taken Rx Past Month', 'Number Days Taken Rx',
       'Number of Rx Meds', 'Gender', 'Age at Screening', 'Race',
       'Educational Attainment', 'Marital Status', 'Sample Interview Weight',
       'MEC Sample Weight', 'Family Income to Poverty Ratio',
       'Bedtime Week/Workdays',
       'Waketime Week/Workdays', 'Sleep Hours Week/Workdays',
       'Bedtime Weekends', 'Waketime Weekends', 'Sleep Hours Weekends',
       'Snoring Frequency Past Year',
       'Frequency Snort/Stop Breathing Past Year',
       'Told Doctor About Sleep Issues', 'Feeling Tired During Day Frequency'])
physical_activity_check

Respondent ID  Vigorous Work Activity  Number of Days Vigorous Work  \
0           109263.0                     NaN                           NaN   
1           109264.0                     NaN                           NaN   
2           109265.0                     NaN                           NaN   
3           109266.0                     2.0                           NaN   
4           109267.0                     2.0                           NaN   
...              ...                     ...                           ...   
32957       124821.0                     1.0                           1.0   
32958       124821.0                     1.0                           1.0   
32959       124821.0                     1.0                           1.0   
32960       124822.0                     2.0                           NaN   
32961       124822.0                     2.0                           NaN   

       Minutes of Vigorous Work Daily  Moderate Work Activity  \
0                                 NaN                     NaN   
1                                 NaN                     NaN   
2                                 NaN                     NaN   
3                                 NaN                     2.0   
4                                 NaN                     2.0   
...                               ...                     ...   
32957                            10.0                     2.0   
32958                            10.0                     2.0   
32959                            10.0                     2.0   
32960                             NaN                     1.0   
32961                             NaN                     1.0   

       Number of Days Moderate Work  Minutes of Moderate Work Daily  \
0                               NaN                             NaN   
1                               NaN                             NaN   
2                               NaN                             NaN   
3                               NaN                             NaN   
4                               NaN                             NaN   
...                             ...                             ...   
32957                           NaN                             NaN   
32958                           NaN                             NaN   
32959                           NaN                             NaN   
32960                           3.0                           180.0   
32961                           3.0                           180.0   

       Walk or Bike Travel  Number of Days Walk or Bike Weekly  \
0                      NaN                                 NaN   
1                      NaN                                 NaN   
2                      NaN                                 NaN   
3                      2.0                                 NaN   
4                      2.0                                 NaN   
...                    ...                                 ...   
32957                  2.0                                 NaN   
32958                  2.0                                 NaN   
32959                  2.0                                 NaN   
32960                  2.0                                 NaN   
32961                  2.0                                 NaN   

       Daily Minutes Walking or Biking  Vigorous Recreational Activity  \
0                                  NaN                             NaN   
1                                  NaN                             NaN   
2                                  NaN                             NaN   
3                                  NaN                             1.0   
4                                  NaN                             1.0   
...                                ...                             ...   
32957                              NaN                             2.0   
32958                              NaN             

In [40]:
PA_remove_dups = physical_activity_check.drop_duplicates(subset=['Respondent ID'], keep='last')
PA_remove_dups

Respondent ID  Vigorous Work Activity  Number of Days Vigorous Work  \
0           109263.0                     NaN                           NaN   
1           109264.0                     NaN                           NaN   
2           109265.0                     NaN                           NaN   
3           109266.0                     2.0                           NaN   
4           109267.0                     2.0                           NaN   
...              ...                     ...                           ...   
32953       124818.0                     2.0                           NaN   
32954       124819.0                     NaN                           NaN   
32955       124820.0                     NaN                           NaN   
32959       124821.0                     1.0                           1.0   
32961       124822.0                     2.0                           NaN   

       Minutes of Vigorous Work Daily  Moderate Work Activity  \
0                                 NaN                     NaN   
1                                 NaN                     NaN   
2                                 NaN                     NaN   
3                                 NaN                     2.0   
4                                 NaN                     2.0   
...                               ...                     ...   
32953                             NaN                     2.0   
32954                             NaN                     NaN   
32955                             NaN                     NaN   
32959                            10.0                     2.0   
32961                             NaN                     1.0   

       Number of Days Moderate Work  Minutes of Moderate Work Daily  \
0                               NaN                             NaN   
1                               NaN                             NaN   
2                               NaN                             NaN   
3                               NaN                             NaN   
4                               NaN                             NaN   
...                             ...                             ...   
32953                           NaN                             NaN   
32954                           NaN                             NaN   
32955                           NaN                             NaN   
32959                           NaN                             NaN   
32961                           3.0                           180.0   

       Walk or Bike Travel  Number of Days Walk or Bike Weekly  \
0                      NaN                                 NaN   
1                      NaN                                 NaN   
2                      NaN                                 NaN   
3                      2.0                                 NaN   
4                      2.0                                 NaN   
...                    ...                                 ...   
32953                  2.0                                 NaN   
32954                  NaN                                 NaN   
32955                  NaN                                 NaN   
32959                  2.0                                 NaN   
32961                  2.0                                 NaN   

       Daily Minutes Walking or Biking  Vigorous Recreational Activity  \
0                                  NaN                             NaN   
1                                  NaN                             NaN   
2                                  NaN                             NaN   
3                                  NaN                             1.0   
4                                  NaN                             1.0   
...                                ...                             ...   
32953                              NaN                             2.0   
32954                              NaN             

In [41]:
#check one column count
PA_remove_dups["Vigorous Work Activity"].count()


9693

In [42]:
#check all count of all columns
PA_remove_dups.count()

Respondent ID                           15560
Vigorous Work Activity                   9693
Number of Days Vigorous Work             2418
Minutes of Vigorous Work Daily           2402
Moderate Work Activity                   9693
Number of Days Moderate Work             4187
Minutes of Moderate Work Daily           4164
Walk or Bike Travel                      9693
Number of Days Walk or Bike Weekly       2253
Daily Minutes Walking or Biking          2241
Vigorous Recreational Activity           9693
Number of Days Vigorous Recreation       2422
Minutes of Vigorous Recreation Daily     2417
Moderate Recreational Activity           9693
Number of Days Moderate Recreation       3904
Minutes of Moderate Recreation Daily     3893
Minutes of Sedentary Activity Daily      9676
dtype: int64

counts are correct for demographic and physical activity data sets in comparison with what is published on CDC website